adapter/internal/ga/client.go

```diff
@@ -155,7 +156,11 @@ func watchAPIs() {
 		} else {
 			lastReceivedResponse = discoveryResponse
 			logger.LoggerGA.Debugf("Discovery response is received : %s", discoveryResponse.VersionInfo)
-			addAPIToChannel(discoveryResponse)
+			if conf.GlobalAdapter.Enabled && conf.ControlPlane.DynamicEnvironments.Enabled {
+				addAPIWithEnvToChannel(discoveryResponse)
+			} else {
+				addAPIToChannel(discoveryResponse)
```

Comment:
Shall we keep a TODO statement here and then delete these unnecessary changes once the fix is rolled out.

Reply:
addressed via https://github.com/wso2/product-microgateway/commit/ae2c06882353bda7a59101381ef3e82f88667a21